packages/components/src/public/QueryColumn.ts

```diff
@@ -42,29 +44,46 @@ export class QueryLookup extends Record({
     declare schemaQuery: SchemaQuery;
     // declare table: string; -- NOT ALLOWING -- USE queryName
 
-    static create(rawLookup): QueryLookup {
-        return new QueryLookup(
-            Object.assign({}, rawLookup, {
-                schemaQuery: new SchemaQuery(rawLookup.schemaName, rawLookup.queryName, rawLookup.viewName),
-            })
+    constructor(rawLookup: Record<string, any>) {
+        Object.assign(this, rawLookup, {
+            schemaQuery: new SchemaQuery(rawLookup.schemaName, rawLookup.queryName, rawLookup.viewName),
+        });
+    }
+
+    hasQueryFilters(operation?: Operation): boolean {
+        return (
+            isAllSamplesSchema(this.schemaQuery) ||
+            (operation !== undefined && this.getFilterGroup(operation) !== undefined)
         );
     }
 
-    hasQueryFilters(): boolean {
-        return isAllSamplesSchema(this.schemaQuery);
+    getFilterGroup(operation: Operation): FilterGroup {
+        return this.filterGroups?.find((filterGroup: FilterGroup) => filterGroup.operation === operation);
```

Comment:
nit: The typings will be known within the processing function so no need to declare `filterGroup: FilterGroup` as `this.filterGroup` is already established as `FilterGroup[]`.